File: 506-relative-ranks/relative-ranks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        int N = score.length;

        PriorityQueue<Pair<Integer, Integer>> maxHeap = new PriorityQueue<>((a,b) -> b.getKey() - a.getKey());

        for (int i = 0; i < N; i++) {
            maxHeap.offer(new Pair(score[i], i));
        }

        String[] rank = new String[N];
        int place = 1;
        while (!maxHeap.isEmpty()) {
            int index = maxHeap.poll().getValue();
            if (place == 1) {
                rank[index] = "Gold Medal"; 
            } else if (place == 2) {
                rank[index] = "Silver Medal";
            } else if (place == 3) {
                rank[index] = "Bronze Medal";
            } else {
                rank[index] = String.valueOf(place);
            }
            place++;
        }
        return rank;

    }
}

File: 506-relative-ranks/relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        sortedScore = sorted(score, reverse = True)
        rank = {}

        for i in range(len(sortedScore)):
            s = sortedScore[i]
            if i == 0:
                rank[s] = "Gold Medal"
            elif i == 1:
                rank[s] = "Silver Medal"
            elif i == 2:
                rank[s] = "Bronze Medal"
            else:
                rank[s] = str(i + 1)

        ans = []
        for i in range(len(score)):
            ans.append(rank[score[i]])

        return ans